### PR TITLE
Fix 4327 - make float_str_to_int_string append warning

### PR DESCRIPTION
**Pull request by liufuyang**
## What have you changed? (mandatory)

Make function float_str_to_int_string stop returning error
when input float string cannot be casted into an int string,
instead it now returns the original float string but with
warning attached.

See https://github.com/tikv/tikv/issues/4327

## What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

- Bug fix (change which fixes an issue)

## How has this PR been tested? (mandatory)

Unit tests done.

## Does this PR affect documentation (docs) update? (mandatory)

If there is document change, please file a PR in ([docs](https://github.com/tikv/tikv/tree/master/docs)) and add the PR number here.

(I am not sure...)

### CODE REVIEW

**Comment by liufuyang** [comment]
@breeswish Hey Wish, can you give a look at this PR to see if it looks like that I am going to the correct direction?

Also, I am not sure, when attaching warnings, I should use `Error::overflow` or `Error::truncated_wrong_val`... From the issue page you can see the MySQL5.7's warning is something like `Warning | 1292 | Truncated incorrect INTEGER value: '1e10'` perhaps I should use `Error::truncated_wrong_val`? But then it feels weird when reading the code and the warning is attached when it overflows i64::MAX... 🤔

File: src/coprocessor/codec/convert.rs [hunk 1]
```diff
@@ -463,16 +472,25 @@ mod tests {
...
+        // Firstly, make sure no error returns, instead a valid float string is returned
         for i in cases {
-            let o = super::float_str_to_int_string(i);
-            assert!(o.is_err());
+            let o = super::get_valid_int_prefix(&mut ctx, i);
+            assert_eq!(o.unwrap(), i);
+        }
+
+        // Secondly, make sure warnings are attached when the float string cannot be casted to a valid int string
+        assert_eq!(ctx.take_warnings().warnings.len(), 2);
```
Comment (breeswish) [inline src/coprocessor/codec/convert.rs]: When warnings are taken, you cannot take it again (which will break Line 486). You can consider storing warnings you took in another variable.

Comment (breeswish) [inline src/coprocessor/codec/convert.rs]: When warnings are taken, you cannot take it again (which will break Line 486). You can consider storing warnings you took in another variable.

File: src/coprocessor/codec/convert.rs [hunk 2]
```diff
@@ -463,16 +472,25 @@ mod tests {
...
+        for warning in ctx.take_warnings().warnings {
```
Comment (breeswish) [inline src/coprocessor/codec/convert.rs]: The warnings are already taken, so this loop never loops.

**Comment by liufuyang** [comment]
@breeswish Your input is addressed :)

**Comment by breeswish** [comment]
Thanks a lot!

**Pull request merged**